src/app/filesystem/permanent-filesystem.ts

```diff
@@ -5,16 +5,38 @@ import {
   FolderIdentifier,
   RecordIdentifier,
 } from './types/filesystem-identifier';
+import { FolderCache } from './folder-cache';
 
 export class PermanentFilesystem {
-  constructor(private api: FilesystemApi) {}
+  private folderCache: FolderCache;
+
+  constructor(private api: FilesystemApi, private cache?: FolderCache) {
+    if (cache) {
+      this.folderCache = cache;
+    } else {
+      this.folderCache = new FolderCache();
+    }
+  }
 
   public async getArchiveRoot(archive: ArchiveIdentifier): Promise<FolderVO> {
     return await this.api.getRoot(archive);
   }
 
   public async getFolder(folder: FolderIdentifier): Promise<FolderVO> {
-    return await this.api.navigate(folder);
+    const cachedFolder = this.folderCache.getFolder(folder);
+    if (cachedFolder) {
+      setTimeout(() => {
+        this.fetchFolderFromApi(folder);
+      }, 0);
```

Comment:
How does this change the value returned by `getFolder` from the cached value returned originally to the result of the API call? (I can see that it does, because the test passes, but I don't understand it).